iree/compiler/InputConversion/MHLO/MHLOToMHLOPreprocessing.cpp

```diff
@@ -674,6 +674,81 @@ class ScatterRank0Value : public OpRewritePattern<mhlo::ScatterOp> {
   }
 };
 
+// Mhlo of non-finite values (e.g. NaN, inf) and 0.0 produce 0.0 for XLA. For
+// linalg we need to conver these to select operations.
+class MulCastOfBool : public OpRewritePattern<mhlo::MulOp> {
+ public:
+  using OpRewritePattern<mhlo::MulOp>::OpRewritePattern;
+
+  // Traverse upward past common operations to see if the value came from a
+  // boolean tensor.
+  static bool IsFromBool(Value val) {
```

Comment:
Not a fan of `static` class methods in patterns. I'd bias towards a static method outside of the class. Also naming should be `isFromBool`